tests/codegen/range-attribute.rs

```diff
@@ -0,0 +1,33 @@
+// Checks that range metadata gets emitted on functions result and arguments
+// with scalar value.
```

Comment:
Some more tests that would be nice:
- slice length (I don't think that will get range attributes yet, but when it does, it'll resolve #106737. Having a test is still useful whether it gets the attribute or not, just to show the current state.)
- discriminant of an enum with a payload (one that gets scalarpair layout) e.g. `enum Foo { A(u64), B(u64), C(u64) }`.

Reply:
think that I have added the test that you wanted. I'm letting someone more familiar create a PR for adding the range on slices.